screens/AppNavigator/screens/Dex/DexAddLiquidity.tsx

```diff
@@ -0,0 +1,191 @@
+import { AddressToken } from '@defichain/whale-api-client/dist/api/address'
+import { PoolPairData } from '@defichain/whale-api-client/dist/api/poolpair'
+import { StackScreenProps } from '@react-navigation/stack'
+import BigNumber from 'bignumber.js'
+import * as React from 'react'
+import { useCallback, useState } from 'react'
+import { TouchableOpacity } from 'react-native'
+import { ScrollView } from 'react-native-gesture-handler'
+import tailwind from 'tailwind-rn'
+import { Text, TextInput, View } from '../../../../components'
+import { getTokenIcon } from '../../../../components/icons/tokens/_index'
+import { PrimaryColorStyle } from '../../../../constants/Theme'
+import { useTokensAPI } from '../../../../hooks/wallet/TokensAPI'
+import { DexParamList } from './DexNavigator'
+
+type Props = StackScreenProps<DexParamList, 'AddLiquidity'>
+type EditingAmount = 'primary' | 'secondary'
+
+interface ExtPoolPairData extends PoolPairData {
+  aSymbol: string
+  bSymbol: string
+  aToBRate: BigNumber
+  bToARate: BigNumber
+}
+
+export function AddLiquidityScreen (props: Props): JSX.Element {
+  // TODO: poll for PoolPairData periodically
+
+  // this component state
+  const [tokenAAmount, setTokenAAmount] = useState<BigNumber>(new BigNumber(0))
+  const [tokenBAmount, setTokenBAmount] = useState<BigNumber>(new BigNumber(0))
+  const [sharePercentage, setSharePercentage] = useState<BigNumber>(new BigNumber(0))
+
+  // gather required data
+  const tokens = useTokensAPI()
+  const { pair: poolPairData } = props.route.params
+  const [aSymbol, bSymbol] = poolPairData.symbol.split('-')
+  const pair: ExtPoolPairData = {
+    ...poolPairData,
+    aSymbol,
+    bSymbol,
+    aToBRate: new BigNumber(poolPairData.tokenB.reserve).div(poolPairData.tokenA.reserve),
```

Comment:
should this be tokenA to tokenB?

Reply:
Oops. Actually this row is correct, next row is the wrong one. Price should be inversely proportional to reserve amount and it make sense, eg: 1 BTC => 1 * 10000DFI / 1 BTC = 10000 DFI